src/PrestaShopBundle/Controller/Admin/ProductImageController.php

```diff
@@ -0,0 +1,189 @@
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2015 PrestaShop SA
+ * @license   http://opensource.org/licenses/osl-3.0.php Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShopBundle\Controller\Admin;
+
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\Validator\Constraints as Assert;
+use PrestaShopBundle\Form\Admin\Type\TranslateType;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Template;
+
+/**
+ * Admin controller for product image
+ */
+class ProductImageController extends FrameworkBundleAdminController
+{
+    /**
+     * Manage upload for product image
+     *
+     * @param int $idProduct
+     * @param Request $request
+     *
+     * @return string
+     */
+    public function uploadImageAction($idProduct, Request $request)
+    {
+        $response = new JsonResponse();
+        $adminProductWrapper = $this->container->get('prestashop.adapter.admin.wrapper.product');
+        $translator = $this->container->get('prestashop.adapter.translator');
+        $return_data = [];
+
+        if ($idProduct == 0 || !$request->isXmlHttpRequest()) {
+            return $response;
+        }
+
+        $form = $this->createFormBuilder(null, array('csrf_protection' => false))
+            ->add('file', 'file', array(
+                'error_bubbling' => true,
+                'constraints' => [
+                    new Assert\NotNull(array('message' => $translator->trans('Please select a file', [], 'AdminProducts'))),
+                    new Assert\Image(array('maxSize' => '8M')),
+                ]
+            ))
+            ->getForm();
+
+        $form->handleRequest($request);
+
+        if ($request->isMethod('POST')) {
+            if ($form->isValid()) {
+                $return_data = $adminProductWrapper->getInstance()->ajaxProcessaddProductImage($idProduct, 'form', false)[0];
+            } else {
+                $error_msg = array();
+                foreach ($form->getErrors() as $key => $error) {
+                    $error_msg[] = $error->getMessage();
+                }
+                $return_data = array('message' => implode(" ", $error_msg));
+                $response->setStatusCode(400);
+            }
+        }
+
+        return $response->setContent(json_encode($return_data));
```

Comment:
this will probably break your response as the json_encode is already done (https://github.com/symfony/http-foundation/blob/master/JsonResponse.php#L96) or is it intended ?